src/test/kotlin/com/exactpro/th2/lwdataprovider/TestSseEventSearchRequest.kt

```diff
@@ -1,93 +1,162 @@
+/*******************************************************************************
+ * Copyright 2021-2021 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
 package com.exactpro.th2.lwdataprovider
 
 import com.exactpro.cradle.TimeRelation
+import com.exactpro.th2.common.grpc.EventID
 import com.exactpro.th2.dataprovider.grpc.EventSearchRequest
 import com.exactpro.th2.lwdataprovider.entities.exceptions.InvalidRequestException
 import com.exactpro.th2.lwdataprovider.entities.requests.SseEventSearchRequest
+import com.exactpro.th2.lwdataprovider.entities.requests.SseMessageSearchRequest
 import com.google.protobuf.Timestamp
 import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Nested
 import org.junit.jupiter.api.Test
 import org.junit.jupiter.api.assertThrows
 
 class TestSseEventSearchRequest {
-    @Test
-    fun testConstructorParam(){
-        // startTimestamp - 1, endTimestamp - 2, searchDirection - AFTER (default)
-        var params = mutableMapOf("startTimestamp" to listOf("1"),
-            "endTimestamp" to listOf("2", "3"))
-        var eventSearchReq = SseEventSearchRequest(params)
-        Assertions.assertNotNull(eventSearchReq.startTimestamp)
-        Assertions.assertNotNull(eventSearchReq.endTimestamp)
-        Assertions.assertEquals(TimeRelation.AFTER, eventSearchReq.searchDirection)
 
-        // startTimestamp and resumeFromId - nulls
-        params.remove("startTimestamp")
-        params.remove("endTimestamp")
-        assertThrows<InvalidRequestException> {
-            eventSearchReq = SseEventSearchRequest(params)
+    @Nested
+    inner class TestConstructorParam(){
+        @Test
+        // when startTimestamp and resumeFromId - nulls, should throw exception
+        fun testEmptyParamsMap(){
+            assertThrows<InvalidRequestException> {
+                SseEventSearchRequest(mapOf())
+            }
         }
 
-        // resumeFromId != null - testing checkEndTimestamp validation from here till the end
-        params["resumeFromId"] = listOf("1")
-        eventSearchReq = SseEventSearchRequest(params)
-        Assertions.assertNull(eventSearchReq.startTimestamp)
-        Assertions.assertNull(eventSearchReq.endTimestamp)
+        @Test
+        // resumeFromId != null, startTimestamp and endTimestamp - nulls
```

Comment:
Comments should go before the annotations
```suggestion
        // resumeFromId != null, startTimestamp and endTimestamp - nulls
        @Test
```